pages/dashboard/index.tsx

```diff
@@ -1,42 +1,73 @@
-import * as React from 'react'
-import { signIn, useSession } from 'next-auth/client'
-import { apiClient } from '../../utils.client'
-import { useMutation, useQuery } from 'react-query'
-import { Box, Button, Container, Flex, FormControl, FormLabel, Heading, HStack, Input, Link, LinkBox, LinkOverlay, Menu, MenuButton, MenuDivider, MenuGroup, MenuItem, MenuList, Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay, Skeleton, Spacer, Stack, Tag, toast, useDisclosure, useToast, VStack } from '@chakra-ui/react'
-import { useForm } from 'react-hook-form'
-import { Project } from '@prisma/client'
-import { ChevronDownIcon } from '@chakra-ui/icons'
-import type { UserSession } from '../../service'
-import { useRouter } from 'next/router'
-import { Head } from '../../components/Head'
-import { Footer } from '../../components/Footer'
-import { Navbar } from '../../components/Navbar'
-import { getSession } from '../../utils.server'
-
-export const createProject = async (body: {
-  title: string
-}) => {
-  const res = await apiClient.post('/projects', {
-    title: body.title
+import * as React from "react"
+import { signIn, useSession } from "next-auth/client"
+import { apiClient } from "../../utils.client"
+import { useMutation, useQuery } from "react-query"
+import {
+  Box,
+  Button,
+  Container,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  HStack,
+  Input,
+  Link,
+  LinkBox,
+  LinkOverlay,
+  Menu,
+  MenuButton,
+  MenuDivider,
+  MenuGroup,
+  MenuItem,
+  MenuList,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Skeleton,
+  Spacer,
+  Stack,
+  Tag,
+  toast,
+  useDisclosure,
+  useToast,
+  VStack,
+} from "@chakra-ui/react"
+import { useForm } from "react-hook-form"
+import { Project } from "@prisma/client"
+import { ChevronDownIcon } from "@chakra-ui/icons"
+import type { UserSession } from "../../service"
+import { useRouter } from "next/router"
+import { Head } from "../../components/Head"
+import { Footer } from "../../components/Footer"
+import { Navbar } from "../../components/Navbar"
+import { getSession } from "../../utils.server"
+
+export const createProject = async (body: { title: string }) => {
+  const res = await apiClient.post("/projects", {
+    title: body.title,
   })
   return res.data
 }
 
 export const getAllProjects = async () => {
-  const res = await apiClient.get('/projects')
+  const res = await apiClient.get("/projects")
   return res.data.data
 }
 
-function Dashboard(props: {
-  session: UserSession
-}) {
-
+function Dashboard(props: { session: UserSession }) {
   if (!props.session) {
-    signIn()
-    return ''
+    React.useEffect(() => {
```

Comment:
I think the `if` statement should inside useEffect https://reactjs.org/docs/hooks-effect.html

Reply:
you already use hooks outside the if block, it's actually fine here cause `props.session` never changes once the page is mounted.